csvcubed/csvcubed/schema/codelist-config/v1_0/schema.json

```diff
@@ -0,0 +1,155 @@
+  "type": "object",
+  "required": [
+    "$schema",
+    "title",
+    "description",
+    "license",
+    "concepts"
+  ],
+  "properties": {
+    "$schema": {
+      "type": "string",
+      "default": "https://raw.githubusercontent.com/GSS-Cogs/csvcubed/294-define-codelist-in-qubeconfig/csvcubed/csvcubed/schema/codelist-config/v1_0/schema.json"
+    },
+    "id": {
+      "description": "A unique id for the code list.",
+      "type": "string"
+    },
+    "title": {
+      "description": "A title for the data set.",
+      "type": "string"
+    },
+    "description": {
+      "description": "A long description of the code list.",
+      "type": "string"
+    },
+    "summary": {
+      "description": "A short summary of the code list.",
+      "type": "string"
+    },
+    "creator": {
+      "description": "The creator of the data set (uri).",
+      "$ref": "http://purl.org/csv-cubed/resources/organisations.json#/uris"
+    },
+    "publisher": {
+      "description": "The publisher of the data set (uri).",
+      "$ref": "http://purl.org/csv-cubed/resources/organisations.json#/uris"
+    },
+    "themes": {
+      "$ref": "#/definitions/v1.0/themeTypes"
+    },
+    "keywords": {
+      "$ref": "#/definitions/v1.0/keywordTypes"
+    },
+    "dataset_issued": {
+      "description": "Date time that the data set was initially published in ISO 8601 format, e.g. 2022-03-31 or 2022-03-31T12:54:30Z.",
+      "type": "string"
+    },
+    "dataset_modified": {
+      "description": "Date time that the data set was last modified published in ISO 8601 format, e.g. 2022-03-31 or 2022-03-31T12:54:30Z.",
+      "type": "string"
+    },
+    "license": {
+      "description": "URI that represents the copyright license applying to this codelist.",
+      "$ref": "http://purl.org/csv-cubed/resources/licenses.json#/uris"
+    },
+    "concepts": {
+      "$ref": "#/definitions/v1.0/concepts"
+    }
+  },
+  "definitions": {
+    "v1.0": {
+      "concepts": {
+        "description": "List of concepts in the codelist.",
+        "type": "array",
+        "uniqueItems": true,
+        "minItems": 1,
+        "items": {
+          "$ref": "#/definitions/v1.0/concept"
+        }
+      },
+      "concept": {
+        "description": "Concept in the codelist.",
+        "type": "object",
+        "required": [
+          "label",
+          "notation"
+        ],
+        "properties": {
+          "label": {
+            "description": "Label of the concept.",
+            "type": "string"
+          },
+          "description": {
+            "description": "Description of the concept.",
+            "type": "string"
+          },
+          "notation": {
+            "description": "Notation of the concept.",
+            "type": "string"
+          },
+          "sort_order": {
+            "description": "Sort order of the concept. If not provided, the order will be inferred from the concepts list.",
```

Comment:
Sort order should be alphabetical by `["label", "notation"]`, or manually sorted using `sort_order`. I suggest we have a `sort` keyword at the head level of the concept scheme, allowing people to sort by
* ascending (default) or descending
* label (default) or notation

__or__

manually using `sort_order`


Mixing of the two shouldn't be supported.

Reply:
Thanks for the heads up, @GDonRanasinghe .

Just to be clear I think the *default* behaviour should be defining the sort order in the same order in which items are defined in the list, whilst supporting people to explicitly define the sort order or letting them tell us to asc/desc the alphabetical order.

Reply:
Your approach obviates the need for a `sort_order` key, which means that a sort key at the top level is optional. I'm okay with this.